src/plots/plots.js

```diff
@@ -426,6 +426,15 @@ plots.supplyDefaults = function(gd) {
         trace  = newData[i];
 
         fullTrace = plots.supplyDataDefaults(trace, i, newFullLayout);
+
+        // If autobinned before, autobin again. supplyDataDefaults sees
+        // the generated bin object and assumes it has been set manually
+        // so we need to override that behaviour.
+        if (oldFullData[i]){
```

Comment:
hmm, on second thought, moving this block to `restyle` would not fix the problem in `Plotly.extendTraces`